synapse/storage/data_stores/main/events_worker.py

```diff
@@ -189,21 +190,23 @@ def get_events(
     @defer.inlineCallbacks
     def get_events_as_list(
         self,
-        event_ids,
-        check_redacted=True,
-        get_prev_content=False,
-        allow_rejected=False,
+        event_ids,  # type: List[Any]
+        check_redacted=True,  # type: bool
+        get_prev_content=False,  # type: bool
+        allow_rejected=False,  # type: bool
+        allow_redacted=True,  # type: bool
     ):
         """Get events from the database and return in a list in the same order
         as given by `event_ids` arg.
 
         Args:
-            event_ids (list): The event_ids of the events to fetch
-            check_redacted (bool): If True, check if event has been redacted
+            event_ids: The event_ids of the events to fetch
+            check_redacted: If True, check if event has been redacted
                 and redact it.
-            get_prev_content (bool): If True and event is a state event,
+            get_prev_content: If True and event is a state event,
                 include the previous states content in the unsigned field.
-            allow_rejected (bool): If True return rejected events.
+            allow_rejected: If True, return rejected events.
+            allow_redacted: If True, return redacted events.
```

Comment:
could you document the interaction between `check_redacted` and `allow_redacted` more clearly? What is the meaning of `check_redacted` if `allow_redacted` is False?

(maybe we should replace `check_redacted` with a tristate?)

Reply:
Indeed, I've switched it out for an enum now, let me know what you think.